src/Nethermind/Nethermind.Core/Attributes/Metrics.cs

```diff
@@ -16,3 +16,14 @@ public sealed class CounterMetricAttribute : Attribute { }
 /// </summary>
 [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
 public sealed class GaugeMetricAttribute : Attribute { }
+
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+public sealed class KeyIsLabel : Attribute
```

Comment:
Do we need a custom attribute here? `DisplayNameAttribute` isn't enough?

Even if I don't get the type name then. Also Should have `Attribute` suffix based on .net conventions.

Reply:
No, `DisplayNameAttribute` change the name/metrics name. This specify that the key of the dictionary is used as a label instead of creating a new metrics for each key like we currently do.